wordpress-mobile/WordPress-iOS
# UserPersistentStore: Refactor & Replace existing `standard` calls

Replaces `UserDefaults.standard` calls with new `UserPersistentStoreFactory.instance()` calls. Latter will use the feature flag to return either standard or the shared instance.

When we enable the feature flag, the standard dictionary is moved to the shared one and the standard is wiped. But `standard` is passed in the `FeatureFlagOverrideStore` as `init(store: KeyValueDatabase = UserDefaults.standard)`. If I use Factory there, it will cause a cycler dependency as the Factory uses `FeatureFlag.sharedUserDefaults` to determine which class to use. As a result, when we enable the flag, in next build the flag will be disabled again as the `standard` will be wiped.
Also if one enables and then disables the Feature Flag, all UserDefaults will be lost of disabled state as we wipe it to not have duplicate data. This won't be a concern for users once we release there won't be a toggle to go back for them. But for testing it might cause confusion.

I would be happy for any suggestions on this one.

To test:
1. Install & Log in WP.
2. Enable Shared Defaults Feature Flag
3. Modify a persisting user setting like App Theme.
4. Run JP and Log in.
5. Enable the feature flag in JP. (You might need to re-run the JP app after if the default you're checking is only read in launch)
6. Check if changes reflect there.

## Regression Notes
1. Potential unintended areas of impact
Plenty. All the defaults ideally need to be tested. I will do a full test run over the next days. Since the whole thing is behind a Flag, we should be fine until that is defaulted to true, i.e. release. I will post my findings in the PR that enables this.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
See 1.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
